homeassistant/helpers/template.py

```diff
@@ -99,8 +99,23 @@ def extract_entities(
             extraction_final.append(variables["trigger"]["entity_id"])
         elif result.group("entity_id"):
             if result.group("func") == "expand":
-                for entity in expand(hass, result.group("entity_id")):
-                    extraction_final.append(entity.entity_id)
+                group_entity_id = result.group("entity_id")
+                group_entity = _get_state(hass, group_entity_id)
```

Comment:
How do you know that groups have been initialized? :)

```suggestion
                group_entity = hass.states.get(group_entity_id)
```

Reply:
`after_dependencies` doesn't trigger `group` to be set up. It will only load if the user has it referenced in their `configuration.yaml`.

Reply:
Sorry, my example was incomplete. I actually _had_ groups configured, and yet `after_dependendies` didn't trigger groups to be initialized before template sensors.

I've added a draft PR with a test (#36786) to recreate this behavior, it would be great if you could have a look.

Reply:
Rebase this PR and `after_dependencies` will work.

Reply:
I rebased the other PR (see #37766), but it looks like it doesn't make the test pass. Do I test incorrectly for the order of loaded integrations?